### PR TITLE
Added standalone mode that can load multiple different modes

### PR DESCRIPTION
**Pull request by dankar**
This mode adds a way of having multiple standalone modes at the same time, and choosing between them when entering standalone mode. It's at least useful for me to have that.

I have not tried this with all the different modes that exists, but a quick test with a few different ones seems to work fine.

The code is very messy since it's done just with some macros. It would be much cleaner to use some code generation, but this works.

### CODE REVIEW

**Comment by iceman1001** [comment]
Interesting,   not quite how I saw multi standalone modes would look like.   A different solution indeed.

**Comment by dankar** [comment]
> Interesting, not quite how I saw multi standalone modes would look like. A different solution indeed.

Feel free to ignore this PR if a more robust way is incoming. It was just a quick useful hack. :)

**Comment by iceman1001** [comment]
I haven't seen anyone working on switching standalone modes functionality for a long time.

File: armsrc/Standalone/dankarmulti.c [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+#define MODE_NAME sniff14a
+#define MODE_FILE "hf_14asniff.c"
+#include "dankarmulti.h"
+#define MODE_NAME em4100
+#define MODE_FILE "lf_em4100rswb.c"
+#include "dankarmulti.h"
+#define MODE_NAME icehid
+#define MODE_FILE "lf_icehid.c"
+#include "dankarmulti.h"
```
Comment (iceman1001) [inline armsrc/Standalone/dankarmulti.c:66]: question,
how does re-defining the same macro work in the add_mode down below?

Comment (dankar) [reply armsrc/Standalone/dankarmulti.c:66]: The same name is used again for ADD_MODE simply to add the different modes to a list. It simplifies enumerating the included modes.

Comment (iceman1001) [reply armsrc/Standalone/dankarmulti.c:66]: yes,  but the redefine is done before the ADD_MODE macro..

Comment (dankar) [reply armsrc/Standalone/dankarmulti.c:66]: The macros are undef:d in the "dankarmulti.h" header, at the end of it, meaning they can be redefined for the next mode.

Comment (iceman1001) [reply armsrc/Standalone/dankarmulti.c:66]: .... sneaky way of dealing with macro defines...   Not sure about it,

**Comment by iceman1001** [comment]
@doegox  what do you think?   Shall we merge or not?   I am skeptical to the "innovative" way of using macros,   then on the other hand...  bleeding edge.

@dankar    You think you can comment your code to explain how those macros works,  so it doesn't confuse ppl like me?

**Comment by doegox** [comment]
Imho we can merge :)

**Comment by dankar** [comment]
Give me a day and I can add to the comments about what it actually does :)

**Comment by iceman1001** [comment]
You can make a new PR with comments :)

Thanks!

**Pull request merged**